networkx/drawing/layout.py

```diff
@@ -1463,15 +1465,16 @@ def estimate_factor(n, swing, traction, speed, speed_efficiency, jitter_toleranc
         repulsion = np.einsum("ijk, ij -> ik", diff, factor)
 
         # gravity
-        gravities = (
-            -gravity
-            * mass[:, None]
-            * pos_arr
-            / np.linalg.norm(pos_arr, axis=-1)[:, None]
-        )
-
+        pos_centered = pos_arr - np.mean(pos_arr, axis=0)
```

Comment:
avoid recomputing this in the else clause below. Also gives a hopefully helpful name to the expression.